packages/snap/src/rpc/signMessage.ts

```diff
@@ -0,0 +1,77 @@
+import bitcoinMessage from 'bitcoinjs-message';
+import {SnapError, RequestErrors} from '../errors';
+import {BitcoinNetwork, ScriptType, Snap} from '../interface';
+import {getPersistedData} from '../utils';
+import {heading, panel, text, divider} from '@metamask/snaps-ui';
+import {pathMap} from '../rpc/getExtendedPublicKey';
+import {getNetwork} from '../bitcoin/getNetwork';
+import * as bitcoin from 'bitcoinjs-lib';
+import { getHDNode } from '../utils/getHDNode';
+
+
+export const signMessage = async (
+  domain: string,
+  snap: Snap,
+  message: string,
+  // TODO: implement bip322 message signing
+  protocol: 'ecdsa' | 'bip322' = 'ecdsa',
+) => {
+  if (protocol !== 'ecdsa') {
+    throw SnapError.of(RequestErrors.ActionNotSupport);
+  }
+
+  const snapNetwork = await getPersistedData<BitcoinNetwork>(
+    snap,
+    'network',
+    '' as BitcoinNetwork,
+  );
+
+  const btcNetwork = getNetwork(snapNetwork);
+  const path = [...pathMap[ScriptType.P2SH_P2WPKH], "0'", '0', '0'];
```

Comment:
Instead of hardcode the path here, it should be passed in as the parameter.

Reply:
Could you please elaborate on this? How do you see it?

Reply:
I mean add one parameter on the signMessage function

Reply:
The PSBT data includes the BIP32 path in the inputs field to indicate which key to use for signing. For signMessage, this information should also be included to specify which key is being used for signing.